trie-db/src/proof/generate.rs

```diff
@@ -480,28 +574,14 @@ fn match_key_to_branch_node<'a, 'b, C: NodeCodec>(
 	}
 }
 
-fn value_with_omission<'a>(
-	node_data: &'a [u8],
-	value_range: &Option<Range<usize>>,
-	omit: bool
-) -> Option<&'a [u8]>
-{
-	if omit {
-		None
-	} else {
-		value_range.clone().map(|range| &node_data[range])
-	}
-}
-
 /// Unwind the stack until the given key is prefixed by the entry at the top of the stack. If the
 /// key is None, unwind the stack completely. As entries are popped from the stack, they are
 /// encoded into proof nodes and added to the finalized proof.
-fn unwind_stack<C: NodeCodec>(
-	stack: &mut Vec<StackEntry<C>>,
+fn unwind_stack<L: TrieLayout>(
+	stack: &mut Vec<StackEntry<L::Codec>>,
 	proof_nodes: &mut Vec<Vec<u8>>,
 	maybe_key: Option<&LeftNibbleSlice>,
-) -> TrieResult<(), C::HashOut, C::Error>
-{
+) -> TrieResult<(), TrieHash<L>, CError<L>> {
```

Comment:
those changes look not necessary, is it better usage of types than before ?

Reply:
I did revert them.